transports/webrtc/src/udp_mux.rs

```diff
@@ -194,7 +194,11 @@ impl UDPMuxNewAddr {
 }
 
 fn write_packet_to_conn_from_addr(conn: UDPMuxConn, packet: Vec<u8>, addr: SocketAddr) {
-    tokio::spawn(async move {
+    // Writing the packet should be quick given it just buffers the data (no actual IO).
+    //
+    // Block until completion instead of spawning to provide backpressure to the clients.
+    // NOTE: `block_on` could be removed once/if `write_packet` becomes sync.
+    futures::executor::block_on(async move {
```

Comment:
Say rust-libp2p runs on a machine with a single CPU / OS thread. If I am not mistaken this `block_on` waits for some other future task to read from the buffer / connection. That other task would run on the same thread, though would not be scheduled, given that the only thread is blocked here.

As far as I can tell this would be a deadlock. @melekes am I missing something?

Reply:
>  If I am not mistaken this block_on waits for some other future task to read from the buffer / connection.

what made u think so? it just buffers data in-memory